indico/modules/events/editing/controllers/base.py

```diff
@@ -120,7 +120,7 @@ def _process_args(self):
         self.editable_type = EditableType[request.view_args['type']]
         self.editable = (Editable.query
                          .with_parent(self.contrib)
-                         .filter_by(type=self.editable_type)
+                         .filter_by(type=self.editable_type, is_deleted=False)
```

Comment:
I suggest changing the relationship's join criterion to already exclude deleted editables - that way you never need to filter by it when using with_parent or the relationship directly.